src/app/backend/namespacelist.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+// Returns a list of all namespaces in the cluster.
+func GetNamespaceList(client *client.Client) ([]string, error) {
```

Comment:
You return a list of strings here, instead of a struct instance that has a list inside. GetReplicaSetList api call does the opposite. It returns an object that has a list. This is an inconsistency that we should fix and agree on early in the project.

I am in favor of returning a struct instance here. This is because in client code it is easy to find out who produces this resource, you just need to grep for it. It is also possible to add further properties to the returned struct, which is not possible if you return an array. Think of properties such as "generation", "paginationStart", "timestamp", etc.

See also how Kubernetes code does this. They always return an object that has some auxiliary fields and the array inside. You can also do a research on the internet to check what is recommended. If nothing, then following Kubernetes core is good to me.

What do you think?